webapp/channels/src/components/channel_header_menu/channel_header_menu.tsx

```diff
@@ -0,0 +1,184 @@
+    }
+
+    const pluginItems = pluginMenuItems.map((item) => {
+        return (
+            <Menu.Item
+                id={item.id + '_pluginmenuitem'}
+                key={item.id + '_pluginmenuitem'}
+                onClick={() => {
+                    if (item.action) {
+                        item.action(channel.id);
+                    }
+                }}
+                labels={<span>{item.text}</span>}
+            />
+        );
+    });
+
+    return (
+        <Menu.Container
+            menuButtonTooltip={{
+                text: channelTitle as string,
+            }}
+            menuButton={{
+                id: 'channelHeaderDropdownButton',
+                class: classNames('channel-header__trigger style--none'),
+                children: (
+                    <>
+                        {archivedIcon}
+                        <strong
+                            id='channelHeaderTitle'
+                            className='heading'
+                            role='heading'
+                            aria-level={2}
+                        >
+                            {channelTitle as string}
+                        </strong>
+                        {sharedIcon}
+                        <ChevronDownIcon size={16}/>
+                    </>
+                ),
+                'aria-label': ariaLabel.toLowerCase(),
+            }}
+            menu={{
+                id: 'channelHeaderDropdownMenu',
+            }}
+            transformOrigin={{
+                horizontal: 'left',
+                vertical: 'top',
+            }}
+            anchorOrigin={{
+                vertical: 'bottom',
+                horizontal: 'left',
+            }}
+        >
+            {isDirect && (
+                <ChannelDirectMenu
+                    channel={channel}
+                    user={user}
+                    isMuted={isMuted}
+                    pluginItems={pluginItems}
+                    isFavorite={isFavorite}
+                    isMobile={isMobile || false}
+                />
+            )}
+            {isGroup && (
+                <ChannelGroupMenu
+                    channel={channel}
+                    user={user}
+                    isMuted={isMuted}
+                    pluginItems={pluginItems}
+                    isFavorite={isFavorite}
+                    isMobile={isMobile || false}
+                />
+            )}
+            {(!isDirect && !isGroup) && (
+                <ChannelPublicPrivateMenu
+                    channel={channel}
+                    user={user}
+                    isMuted={isMuted}
+                    pluginItems={pluginItems}
+                    isFavorite={isFavorite}
+                    isMobile={isMobile || false}
+                    isReadonly={isReadonly}
+                    isDefault={isDefault}
+                    isLicensedForLDAPGroups={isLicensedForLDAPGroups}
+                />
+            )}
```

Comment:
Menu component passes additional props to its first child, here it could be any of the above. that needs to be spread on to the underneath `<Menu.Item` for the it to get the initial focus.
See https://github.com/mattermost/mattermost/blob/4b6216f5a4e3c7e74c7359591d88c0bc33ed50a9/webapp/channels/src/components/user_account_menu/user_account_name_menuitem.tsx#L75

documented here https://github.com/mattermost/mattermost/blob/4b6216f5a4e3c7e74c7359591d88c0bc33ed50a9/webapp/channels/src/components/menu/menu_item.tsx#L119-L126